cmd/zgenhostid/zgenhostid.c

```diff
@@ -0,0 +1,146 @@
+/*
+ * CDDL HEADER START
+ *
+ * The contents of this file are subject to the terms of the
+ * Common Development and Distribution License (the "License").
+ * You may not use this file except in compliance with the License.
+ *
+ * You can obtain a copy of the license at usr/src/OPENSOLARIS.LICENSE
+ * or http://www.opensolaris.org/os/licensing.
+ * See the License for the specific language governing permissions
+ * and limitations under the License.
+ *
+ * When distributing Covered Code, include this CDDL HEADER in each
+ * file and include the License file at usr/src/OPENSOLARIS.LICENSE.
+ * If applicable, add the following below this CDDL HEADER, with the
+ * fields enclosed by brackets "[]" replaced with your own identifying
+ * information: Portions Copyright [yyyy] [name of copyright owner]
+ *
+ * CDDL HEADER END
+ */
+
+/*
+ * Copyright (c) 2020, Georgy Yakovlev.  All rights reserved.
+ */
+
+#include <errno.h>
+#include <fcntl.h>
+#include <getopt.h>
+#include <inttypes.h>
+#include <limits.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/stat.h>
+#include <time.h>
+#include <unistd.h>
+
+static void usage(void);
+
+char path[PATH_MAX] = "/etc/hostid";
+
+int
+main(int argc, char **argv)
+{
+	int opt;			// getopt iterator
+	int forcew = 0;			// force  file write
+	uint32_t hostid = 0;		// final hostid value
+	unsigned long int input_i = 0;	// int to hold converted input
+
+	while ((opt = getopt_long(argc, argv, "fo:h?", 0, 0)) != -1) {
+		switch (opt) {
+		case 'f':
+			forcew = 1;
+			break;
+		case 'o':
+			snprintf(path, sizeof (path), "%s", optarg);
+			break;
+		case 'h':
+		case '?':
+			usage();
+		}
+	}
+
+	if (argv[optind]) {
```

Comment:
It might be clearer to assign  `in_s` here to `argv[optind]` and then perform the check.  I don't believe there's any need to make a copy.  For example the following, and then use `strlen()` below.

```c
char *in_s = argv[optind];
if (in_s != NULL) {
}
```

Reply:
good call. done.